url.go

```diff
@@ -37,21 +37,37 @@ type URL struct {
 	Name     string // "wordpress"
 	Revision int    // -1 if unset, N otherwise
 	Series   string // "precise" or "" if unset
+	Channel  string // "development"
 }
 
 var ErrUnresolvedUrl error = fmt.Errorf("charm or bundle url series is not resolved")
 
+// Channels represent different stages in the development of a charm or bundle.
+const (
+	DevelopmentChannel = "development"
+)
+
 var (
-	validSeries = regexp.MustCompile("^[a-z]+([a-z0-9]+)?$")
-	validName   = regexp.MustCompile("^[a-z][a-z0-9]*(-[a-z0-9]*[a-z][a-z0-9]*)*$")
+	validSeries   = regexp.MustCompile("^[a-z]+([a-z0-9]+)?$")
+	validName     = regexp.MustCompile("^[a-z][a-z0-9]*(-[a-z0-9]*[a-z][a-z0-9]*)*$")
+	validChannels = map[string]bool{
```

Comment:
Since there's only one, why not just inline channel == DevelopmentChannel
until there's need for the map?